src/Stratis.SmartContracts.IntegrationTests/SmartContracts/EcRecoverContract.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Stratis.SmartContracts;
+using EcRecover = Stratis.SCL.Crypto.ECRecover;
+
+public class EcRecoverContract : SmartContract
+{
+    public Address ThirdPartySigner
+    {
+        get
+        {
+            return this.State.GetAddress(nameof(this.ThirdPartySigner));
+        }
+        set
+        {
+            this.State.SetAddress(nameof(this.ThirdPartySigner), value);
+        }
+    }
+
+    public EcRecoverContract(ISmartContractState state, Address thirdPartySigner) : base(state)
+    {
+        this.ThirdPartySigner = thirdPartySigner;
+    }
+
+    public bool CheckThirdPartySignature(byte[] message, byte[] signature)
```

Comment:
I am wondering how the signatures used here are being obtained? Since Keccak is being used the normal "signmessage" API would not be the source of these signatures (which uses sha256).

Reply:
I copied this verbatim from #414 which was approved. Can remove if you like as it's not essential to this PR

Reply:
@YakupIpek , can you comment on this?

Reply:
There's a test where you can see how the signatures are obtained. https://github.com/stratisproject/StratisFullNode/pull/506/files/d4d66b1e2096c7240e9835aa0eabf2df342dbbf4#diff-541900b04541e4da4ce35357818872e3452b844670115c3b54688c7cc6f8d352R76

Reply:
It shows a programmatic approach. Was instead wondering how this fits into the user experience.